src/apiController.js

```diff
@@ -0,0 +1,217 @@
+let apiController = {
+
+
+  // Products API calls
+  getProduct: (pages, countNum) => {
+    let page = pages || 1;
+    let count = countNum || 5;
+    let params = {page: page, count: count};
+    axios.get('/products', {params})
+    .then((data) => {
+      console.log(data);
+    })
+    .catch((err) => {
+      console.error('Oh noes ', err);
+    })
+  },
+  //I'm unsure of whether I'm doing this the right way since I'm not sure how exactly the productId is going to be passed on, but I assume something like this.
+  getProductById: (productId) => {
+    axios.get(`/product/${productId}`)
+    .then((data) => {
+      console.log(data);
+    })
+    .catch((err) => {
+      console.error('Oh noes ', err);
+    })
+  },
+
+  getProductStyles: (productId) => {
+    axios.get(`/product/${productId}/styles`)
+    .then((data) => {
+      console.log(data);
+    })
+    .catch((err) => {
+      console.error('Oh noes ', err);
+    })
+  },
+
+  getProductRelated: (productId) => {
+    axios.get(`/product/${productId}/related`)
+    .then((data) => {
+      console.log(data);
+    })
+    .catch((err) => {
+      console.error('Oh noes ', err);
+    })
+  },
+  //Reviews API Calls
+  getReviews: (pages, countNum, sortVal, productId) => {
+    let page = pages || 1;
+    let count = countNum || 5;
+    let sort = sortVal
+    let product_id = productId
+    let params = {page: page, count: count, sort:sort, product_id: product_id};
+    axios.get('/reviews', {params})
+    .then((data) => {
+      console.log(data);
+    })
+    .catch((err) => {
+      console.error('Oh noes ', err);
+    })
+  },
+  getReivewMetaData: (product_id) => {
+    axios.get('/reviews/meta', {product_id})
+    .then((data) => {
+      console.log(data);
+    })
+    .then ((err) => {
+      console.error('Oh noes ', err);
+    })
+  },
+  // For this next call, it's a post that sends back a lot of data so I figure it's probably better to handle the data in the actual jsx file than here so all we need to get passed into the call is an object of the post data. If you'd rather handle that here let me know and I can change up how this function is set up.
+  postReview: (postVal) => {
+    axios.post('/reviews', postVal)
+    .then((response) => {
+      console.log(response);
+    })
+    .catch((err) => {
+      console.error('Oh noes ', err);
+    })
+  },
+  markHelpful: (review_id) => {
+    axios.put(`/reviews/${review_id}/helpful`)
+    .then((response) => {
+      console.log(response);
+    })
+    .catch((err) => {
+      console.error('Oh noes ', err);
+    })
+  },
+  reportReview: (review_id) => {
+    axios.put(`/reviews/${review_id}/report`)
+    .then((response) => {
+      console.log(response);
+    })
+    .catch((err) => {
+      console.error('Oh noes ', err);
+    })
+  },
+  //Questions and Answers API calls
+  getQuestions: (pages, countNum) => {
```

Comment:
Does this function not need a product_id? I dont have access to the api docs right now, but intuitively questions should be associated with specific products I would think